File: Ejercicios Bucles Chotas 02/Ejercicio 2/abraham.py
num_a = 150
num_b = 225

common_divisor = 1
common_divisor_list = []

# Check all nums until reach the min number bewteen both
while common_divisor <= min(num_a, num_b):
    # Set var to save state of divisible. Default False
    divisor_a = False
    divisor_b = False

    if(num_a % common_divisor == 0):
        divisor_a = True
    if(num_b % common_divisor == 0):
        divisor_b = True

    # If both are divisible adding to list of common divisor
    if(divisor_a and divisor_b):
        common_divisor_list.append(common_divisor)

    # Sum up 1 to the possible common divisor
    common_divisor += 1

print(f"Max common divisor of {num_a} and {num_b} is: {max(common_divisor_list)}")

File: Ejercicios Bucles Chotas 02/Ejercicio 2/abraham.java
import java.util.ArrayList;
import java.util.List;

import static java.lang.Math.min;
import static java.util.Collections.max;

public class Main {
    public static void main(String[] args) {

        int numA = 225;
        int numB = 150;

        int commonDivisor = 1;
        List<Integer> commonDivisorList = new ArrayList<>();

        // Check all nums until reach the min number between both
        while (commonDivisor <= min(numA, numB)){
            // Set var to save state of divisible. Default False
            Boolean divisorA = Boolean.FALSE;
            Boolean divisorB = Boolean.FALSE;

            if(numA % commonDivisor == 0){
                divisorA = Boolean.TRUE;
            }
            if(numB % commonDivisor == 0){
                divisorB = Boolean.TRUE;
            }

            // If both are divisible adding to list of common divisor
            if(divisorA && divisorB){
                commonDivisorList.add(commonDivisor);
            }

            // Sum up 1 to the possible common divisor
            commonDivisor += 1;
        }

        System.out.println("Max common divisor of " + numA +" and " +numB + " is: "+ max(commonDivisorList));

    }
}